syntax = "proto3";

import "header.proto";

message PhotoRequest {
    Header header = 1;
    string filename = 2;
}

message PhotoReply {
    Header header = 1;
    bool fileExists = 2;
    string filename = 16;
    bytes data = 17; // binary data of photo
}

message NewGeoPhotoTakenReply {
    Header header = 1;
    bool fileExists = 2;
    string filename = 16;
    bytes data = 17; // binary data of photo
    float latitude = 18;
    float longitude = 19;
    float altitude = 20; 
}

service PhotoService {
    rpc GetPhoto(PhotoRequest) returns (PhotoReply) {}
}

service TakePhotoService {
    rpc TakePhoto(PhotoRequest) returns (NewGeoPhotoTakenReply) {}
}
